llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -971,6 +971,30 @@ FastMathFlags SPIRVToLLVM::getFastMathFlags(SPIRVValue *bv) {
 
   // For floating-point operations, if "FastMath" is enabled, set the "FastMath"
   // flags on the handled instruction
+  if (m_shaderOptions->fastMathFlags) {
+    if (m_shaderOptions->fastMathFlags & FastMathFlags::AllowReassoc)
+      fmf.setAllowReassoc();
+
+    if (m_shaderOptions->fastMathFlags & FastMathFlags::NoNaNs)
+      fmf.setNoNaNs();
+
+    if (m_shaderOptions->fastMathFlags & FastMathFlags::NoInfs)
+      fmf.setNoInfs();
+
+    if (m_shaderOptions->fastMathFlags & FastMathFlags::NoSignedZeros)
+      fmf.setNoSignedZeros();
+
+    if (m_shaderOptions->fastMathFlags & FastMathFlags::AllowReciprocal)
+      fmf.setAllowReciprocal();
+
+    if (m_shaderOptions->fastMathFlags & FastMathFlags::AllowContract)
+      fmf.setAllowContract();
+
+    if (m_shaderOptions->fastMathFlags & FastMathFlags::ApproxFunc)
+      fmf.setApproxFunc();
+
+    return fmf;
+  }
```

Comment:
This code is largely the same as for `SPIRVGenFastMathFlags`. Maybe it'd be better to factor this out into a helper function that takes fast math flags and returns `FastMathFlags`?

Reply:
The code using `SPIRVGenFastMathFlags` will go away soon, I just can’t do it in a single patch as it’s scattered across repositories.

Reply:
Ack.